skel/gulpfile.js

```diff
@@ -1,44 +1,83 @@
+const sass = require('gulp-sass');
+const postcss = require('gulp-postcss');
+const autoprefixer = require('autoprefixer');
+const lost = require('lost');
+const globbing = require('gulp-css-globbing');
+
+// Javascript
+const babel = require('gulp-babel');
+const uglify = require('gulp-uglify');
+const rename = require('gulp-rename');
+
+// Prevents the pipe/stream from breaking on error
+const plumber = require('gulp-plumber');
+
+// Livereload
+const livereload = require('gulp-livereload');
+
+// Locations of scss, javascript and template files
+// Add scss file paths to de the modulesSrc array to trigger scss proccessing on file change
+// e.g. '../../modules/mod_ginger_foundation/lib/css/src'
+const paths = {
+    modulesSrc: [],
+    scssSrc: 'lib/css/src',
+    scssDest: 'lib/css/site',
+    javascriptSrc: 'lib/js/src',
+    javascriptDest: 'lib/js/dist',
+    templateSrc: 'templates'
 };
 
-gulp.task('sass', function () {
-    gulp.src(paths.cssSource + 'screen.scss')
+// Reload page when .tpl file changes
+gulp.task('tpl', () => gulp.src('/').pipe(livereload()));
+
+// Process, minify and prefix scss files
+gulp.task('sass', () => {
+    gulp.src(`${paths.scssSrc}/screen.scss`)
+        .pipe(plumber())
         .pipe(globbing({ extensions: ['.scss'] }))
-        .pipe(sass({outputStyle : 'compressed'}))
+        .pipe(sass({ outputStyle: 'compressed' }))
         .on('error', sass.logError)
         .pipe(postcss([
             lost(),
             autoprefixer('last 2 versions', 'ie > 7')
         ]))
-        .pipe(gulp.dest(paths.cssDestination))
+        .pipe(gulp.dest(paths.scssDest))
         .pipe(livereload());
 });
 
-gulp.task('sass:watch', function () {
-    var modules = paths.modulesSrc.map(function(path) {
-        return path + '/**/*.scss';
-    });
+// Transpiles ES6 -> minifies -> renames javaScript
+gulp.task('js', () => {
+    gulp.src(`${paths.javascriptSrc}/**/*.js`)
+        .pipe(plumber())
+        .pipe(babel({
+            'presets': [
+                ['env', {
+                    'targets': {
+                        'browsers': ['last 2 versions', 'safari >= 7']
+                    }
+                }]
+            ]
+        }))
+        .pipe(uglify())
+        .pipe(rename(path => path.basename += '.min'))
+        .pipe(gulp.dest(paths.javascriptDest))
+        .pipe(livereload())
+});
+
+// Init livereload server and watch scss, javascript and template files for changes
+gulp.task('watch', () => {
+    livereload.listen();
 
-    var watchPaths = [
-        paths.cssSource + '/**/*.scss'
-    ].concat(modules);
+    var scssCombinedPaths = paths.modulesSrc
+        .map(x => `${x}/**/*.scss`)
+        .concat(`${paths.scssSrc}/**/*.scss`);
 
-    gulp.watch(watchPaths, ['sass']);
+    gulp.watch(scssCombinedPaths, { interval: 100 }, ['sass']);
+    gulp.watch(`${paths.templateSrc}/**/*.tpl`, { interval: 100 }, ['tpl']);
+    gulp.watch(`${paths.javascriptSrc}/**/*.js`, { interval: 100 }, ['js']);
```

Comment:
Does `interval` change the behaviour? I _think_ it only matters for polling, we we don’t (or at least shouldn’t) do: we have inotify.

Reply:
It's indeed the polling interval which I had setup to reduce CPU usage. It's something I tested years ago, something that might not be relevant anymore today.

Reply:
Can you test whether it makes any difference? My hypothesis is that it shouldn’t.

Reply:
There is no significant difference in cpu usage by watching multiple paths simultaneously.

Reply:
And about the polling: as it turns out, the interval *does* matter. Gulp 3 *polls* for changes. 😞 Things seem to be better in Gulp 4 (unreleased) and tools like Webpack.